enterprise/utils.py

```diff
@@ -512,3 +512,20 @@ def safe_extract_key(data, key, default=''):
         return data[key]
     else:
         return default
+
+
+def filter_audit_course_modes(enterprise_customer, course_modes):
+    """
+    Filter audit course modes out if the enterprise customer has the 'Hide audit track' flag enabled.
+
+    Arguments:
+        enterprise_customer: The EnterpriseCustomer that the enrollment was created using.
+        course_modes: iterable with dictionaries containing a required 'mode' key
+    """
+    if not enterprise_customer.hide_audit_track:
+        return course_modes
+    return [
+        course_mode
+        for course_mode in course_modes
+        if not course_mode['mode'] in settings.ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES
```

Comment:
On that topic, though, we should generally _not_ be using bare `settings.VARIABLE` calls. Instead, do something like this:

```python
audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit'])
return [course_mode for course_mode in course_modes if not course_mode['mode'] in audit_modes]
```